### PR TITLE
Change encryption

### PR DESCRIPTION
**Pull request by ffoysal**
# This is the breaking change, as we are changing encryption mechanism.

### CODE REVIEW

**Review by ashwathishiva — APPROVED** [review]

File: pkg/qliksense/context_configs.go [hunk 1]
```diff
@@ -106,7 +102,8 @@ func (q *Qliksense) processSecret(ra *api.ServiceKeyValue, rsaPublicKey *rsa.Pub
 		if k8sSecret.Data == nil {
 			k8sSecret.Data = map[string][]byte{}
 		}
-		k8sSecret.Data[ra.Key] = []byte(base64EncodedSecret)
+		// v1.Secret does enconding, so no need to encode again
```
Comment (ffoysal) [inline pkg/qliksense/context_configs.go:105]: @ashwathishiva did you remember double encoding issue

Comment (ashwathishiva) [reply pkg/qliksense/context_configs.go:105]: yes, I do

File: pkg/qliksense/context_configs.go [hunk 2]
```diff
@@ -451,17 +446,13 @@ func (q *Qliksense) PrepareK8sSecret(targetFile string) (string, error) {
...
 	var resultMap = make(map[string][]byte)
 	for k, v := range dataMap {
-		ba, err := b64.StdEncoding.DecodeString(string(v))
-		if err != nil {
-			err := fmt.Errorf("Not able to decode message: %v", err)
-			return "", err
-		}
-		decryptedString, err := api.Decrypt(ba, rsaPrivateKey)
+		//k8s secrets has already base64 decoed value
+		decryptedString, err := api.DecryptData(v, encryptionKey)
```
Comment (ffoysal) [inline pkg/qliksense/context_configs.go:450]: thats why we had to do another decode @ashwathishiva

File: pkg/qliksense/about.go [hunk 3]
```diff
@@ -194,8 +194,10 @@ func getImageList(yamlContent []byte) ([]string, error) {
 		})
 	}
 	var sortedImageList []string
-	for image, _ := range imageMap {
+	for image, v := range imageMap {
```
Comment (andriy-bulynko) [inline pkg/qliksense/about.go:197]: I think that warning suggests that the range expression should be written like this (we can skip mentioning value altogether):
```
for image := range imageMap {
  sortedImageList = append(sortedImageList, image)
}
```

File: pkg/api/encryption.go [hunk 4]
```diff
@@ -65,104 +49,54 @@ func writeContentToFile(keyData []byte, fileName string) error {
...
-		return nil, err
+func GenerateKey() (string, error) {
+	salt := make([]byte, 32)
+	if _, err := rand.Read(salt); err != nil {
+		return "", err
 	}
-	return ct, nil
+	s := fmt.Sprintf("%x", salt)
+	return s, nil
 }
```
Comment (andriy-bulynko) [inline pkg/api/encryption.go:60]: Wow - I don't think I've seen it done like this before. A lot of funky calls. Looks cool! I mean the whole file..

**Review by andriy-bulynko — APPROVED** [review]

**Pull request merged**